rasa/core/brokers/pika.py

```diff
@@ -602,53 +641,24 @@ def _connect(self) -> None:
             self.host, self.username, self.password, self.port
         )
 
-        self.process_queue = self._get_mp_context().Queue()
         self.pika_message_processor = PikaMessageProcessor(
-            parameters,
-            queues=self.queues,
-            get_message=lambda: self.process_queue.get(),
+            parameters, queues=self.queues
         )
-
-        self.process = self._start_pika_process()
+        self.process = self._start_pika_connection_process()
 
     def _get_mp_context(self) -> multiprocessing.context.BaseContext:
         return multiprocessing.get_context(self.MP_CONTEXT)
 
-    def _start_pika_process(self) -> Optional[multiprocessing.Process]:
+    def _start_pika_connection_process(self) -> Optional[multiprocessing.Process]:
         if self.pika_message_processor:
             process = multiprocessing.Process(
-                target=self.pika_message_processor.run, daemon=True
+                target=self.pika_message_processor.run, args=(self.process_queue,)
             )
             process.start()
             return process
 
         return None
 
-    def _publish(self, body: Text, headers: MessageHeaders = None) -> None:
-        if not self.pika_message_processor:
-            self._connect()
-
-        if (
-            self.process and self.process.is_alive()
-        ) or self.should_keep_unpublished_messages:
-            self.process_queue.put((body, headers))
-
-    def is_ready(
```

Comment:
we can return if the process is alive here. It's used by the `exporter` module afaik